zebra-consensus/src/block/subsidy/founders_reward.rs

```diff
@@ -2,22 +2,36 @@
 //!
 //! [7.7]: https://zips.z.cash/protocol/protocol.pdf#subsidies
 
+use num_integer::div_ceil;
+
 use std::convert::TryFrom;
+use std::str::FromStr;
 
 use zebra_chain::{
     amount::{Amount, Error, NonNegative},
     block::Height,
-    parameters::Network,
+    parameters::{Network, NetworkUpgrade::*},
+    transaction::Transaction,
+    transparent::{Address, Output, ScriptForNetwork},
 };
 
 use crate::block::subsidy::general::{block_subsidy, halving_divisor};
-use crate::parameters::subsidy::FOUNDERS_FRACTION_DIVISOR;
+use crate::parameters::subsidy::*;
+
+/// Returns `true` if we are in the founders reward period of the blockchain.
+pub fn founders_reward_active(height: Height, network: Network) -> bool {
+    let canopy_activation_height = Canopy
+        .activation_height(network)
+        .expect("Canopy activation height is known");
+
+    height < canopy_activation_height
+}
 
 /// `FoundersReward(height)` as described in [protocol specification §7.7][7.7]
 ///
 /// [7.7]: https://zips.z.cash/protocol/protocol.pdf#subsidies
 pub fn founders_reward(height: Height, network: Network) -> Result<Amount<NonNegative>, Error> {
-    if halving_divisor(height, network) == 1 {
+    if halving_divisor(height, network) == 1 && founders_reward_active(height, network) {
```

Comment:
This extra check is redundant on mainnet and testnet

```suggestion
    if founders_reward_active(height, network) {
```